packages/sdk/react-client/src/client/useAgentHostingProvider.ts

```diff
@@ -2,36 +2,43 @@
 // Copyright 2023 DXOS.org
 //
 
-import {
-  type AgentHostingProviderClient,
-  DXOSAgentHostingProviderClient,
-  FakeAgentHostingProvider,
-} from '@dxos/client/services';
+import { type Halo } from '@dxos/client/halo';
+import { type AgentHostingProviderClient, AgentManagerClient, FakeAgentHostingProvider } from '@dxos/client/services';
 import { type Config } from '@dxos/config';
 import { log } from '@dxos/log';
 
 /*
  * Interface for invoking the agent hosting provider client.
  * @experimental
  */
-export const useAgentHostingProviderClient = (config: Config): AgentHostingProviderClient | null => {
-  // TODO: Dynamically discover based on runtime config
-  if (
-    config.get('runtime.app.env.DX_ENVIRONMENT') !== 'development' &&
-    localStorage.getItem('dxos.org/shell/features/agentHosting') !== 'true'
-  ) {
-    return null;
-  }
-
+export const useAgentHostingProviderClient = (config: Config, halo: Halo): AgentHostingProviderClient | null => {
   const agentHostingConfig = config.get('runtime.services.agentHosting');
   if (!agentHostingConfig) {
     log.info('no agent hosting configured');
     return null;
   }
+  // TODO: Dynamically discover based on runtime config
+  let agentHostingProviderClient: AgentHostingProviderClient | null = null;
 
-  if (agentHostingConfig.type === 'LOCAL_TESTING') {
-    return new FakeAgentHostingProvider(config);
-  }
+  switch (agentHostingConfig.type) {
+    case 'LOCAL_TESTING':
+      log.info('using FakeAgentHostingProvider');
+      return new FakeAgentHostingProvider(config, halo);
 
-  return new DXOSAgentHostingProviderClient(config);
+    case 'AGENTHOSTING_API':
+      if (localStorage.getItem('dxos.org/shell/features/agentHosting') !== 'true') {
+        log.info('dxos.org/shell/features/agentHosting not enabled');
+        return null;
+      }
+      agentHostingProviderClient = new AgentManagerClient(config, halo);
```

Comment:
Yeah, as a rule of thumb hooks should not contain the `new` keyword outside of a `useRef` or `useState` call’s arguments.

Instead, seems like we need an `AgentHostingContext` which creates the client using the same props as this hook’s current arguments, then this hook simply gets that instance.